pytest_fixtures/component/puppet.py

```diff
@@ -8,7 +8,10 @@
 @pytest.fixture(scope='session')
 def session_puppet_enabled_sat(session_satellite_host):
     """Satellite with enabled puppet plugin"""
-    yield session_satellite_host.enable_puppet_satellite()
+    if session_satellite_host:
+        yield session_satellite_host.enable_puppet_satellite()
+    else:
+        yield
```

Comment:
better to yield nothing if session_satellite_host fails here for this fixture, and test session won't start, right?
```suggestion
    if session_satellite_host:
        yield session_satellite_host.enable_puppet_satellite()
```

Reply:
Yeah, here its yielding None, and that has to be explicit with just `yield`. 

Also, this change is required, not exactly remember now why! But let it be as this does not cause any harm!